server/platform/services/sharedchannel/sync_send_remote.go

```diff
@@ -679,6 +680,65 @@ func (scs *Service) sendProfileImageSyncData(sd *syncData) {
 }
 
 // sendSyncMsgToRemote synchronously sends the sync message to the remote cluster (or plugin).
+// handleChannelNotSharedError processes the case when a remote indicates a channel
+// is no longer shared. It removes the remote from the shared channel locally and,
+// if it was the last remote, completely unshares the channel.
+func (scs *Service) handleChannelNotSharedError(msg *model.SyncMsg, rc *model.RemoteCluster) {
+	logger := scs.server.Log()
+	logger.Log(mlog.LvlSharedChannelServiceDebug, "Remote indicated channel is no longer shared; unsharing locally",
+		mlog.String("remote", rc.Name),
+		mlog.String("channel_id", msg.ChannelId),
+	)
+
+	// Get the SharedChannelRemote record for this channel and remote
+	scr, getErr := scs.server.GetStore().SharedChannel().GetRemoteByIds(msg.ChannelId, rc.RemoteId)
+	if getErr != nil {
+		logger.Log(mlog.LvlSharedChannelServiceError, "Failed to get shared channel remote",
+			mlog.String("remote", rc.Name),
+			mlog.String("channel_id", msg.ChannelId),
+			mlog.Err(getErr),
+		)
+		return
+	}
+
+	// Get channel details for posting the system message
+	channel, channelErr := scs.server.GetStore().Channel().Get(msg.ChannelId, true)
+	if channelErr != nil {
+		logger.Log(mlog.LvlSharedChannelServiceError, "Failed to get channel details",
+			mlog.String("remote", rc.Name),
+			mlog.String("channel_id", msg.ChannelId),
+			mlog.Err(channelErr),
+		)
+		return
+	}
+
+	// Post a system message to notify users that the channel is no longer shared
+	scs.postUnshareNotification(msg.ChannelId, scr.CreatorId, channel, rc.Name)
+
+	// Remove this remote from the shared channel
+	if _, deleteErr := scs.server.GetStore().SharedChannel().DeleteRemote(scr.Id); deleteErr != nil {
+		logger.Log(mlog.LvlSharedChannelServiceError, "Failed to remove remote from shared channel",
+			mlog.String("remote", rc.Name),
+			mlog.String("channel_id", msg.ChannelId),
+			mlog.Err(deleteErr),
+		)
+		return
+	}
+
+	// Directly unshare the channel regardless of whether there are other remotes
```

Comment:
Why aren't we checking if there are other remotes? Shouldn't we keep the channel marked as shared if there are other remotes connected to it?

Reply:
terminology is a bit confusing. The ticket talks about the unsharing between two remote clusters("his channel is no longer shared"), while in the code the "unshare action" is the complete removal of the sharing of the channel (as opposed to the removal of a remote). I updated the code to reuse the `unshareChannelIfNoActiveRemotes` which wraps the final Unshare call.